src/utils/getUserDetails.ts

```diff
@@ -0,0 +1,21 @@
+import { formatUserDetails } from "../utils/formatUserDetails";
+import { RDS_BASE_API_URL } from "../constants/urls";
+
+export async function getUserDetails(discordId: string) {
+  try {
+    const response = await fetch(
+      `${RDS_BASE_API_URL}/users/?dev=true&discordId=${discordId}`
+    );
+
+    if (!response.ok) {
+      throw new Error(`Failed to fetch user details for ${discordId}.`);
+    }
+
+    const data = await response.json();
+    const user = `${JSON.stringify(data)}`;
+    return formatUserDetails(user);
```

Comment:
Instead of returning formatted text, can you return only the API response so we can reuse the data?